src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -253,18 +252,18 @@ void GcsServer::InitGcsActorManager(const GcsInitData &gcs_init_data) {
 
   if (RayConfig::instance().gcs_actor_scheduling_enabled()) {
     RAY_CHECK(gcs_resource_manager_ && gcs_resource_scheduler_);
-    scheduler = std::make_unique<GcsBasedActorScheduler>(
+    gcs_actor_scheduler_ = std::make_shared<GcsBasedActorScheduler>(
         main_service_, gcs_table_storage_->ActorTable(), *gcs_node_manager_, gcs_pub_sub_,
         gcs_resource_manager_, gcs_resource_scheduler_, schedule_failure_handler,
         schedule_success_handler, raylet_client_pool_, client_factory);
   } else {
-    scheduler = std::make_unique<RayletBasedActorScheduler>(
+    gcs_actor_scheduler_ = std::make_shared<RayletBasedActorScheduler>(
         main_service_, gcs_table_storage_->ActorTable(), *gcs_node_manager_, gcs_pub_sub_,
         schedule_failure_handler, schedule_success_handler, raylet_client_pool_,
         client_factory);
   }
   gcs_actor_manager_ = std::make_shared<GcsActorManager>(
-      std::move(scheduler), gcs_table_storage_, gcs_pub_sub_, *runtime_env_manager_,
+      gcs_actor_scheduler_, gcs_table_storage_, gcs_pub_sub_, *runtime_env_manager_,
```

Comment:
should we still keep it unique ptr? We shouldn't use shared ptr unless it's shared across threads (including user thread)

Reply:
You can pass ref or .get()

Reply:
The `gcs_actor_scheduler_` has to be called in `InstallEventListeners`. BTW, in the next PR, the `gcs_actor_scheduler_` will be called even more (for `JobFinish` and `WorkerDead`, etc.) in that function.

> You can pass ref or .get()
`GcsActorManager` constructor's first argument is not a reference of smart ptr, and passing a unique_ptr.get() also seems a little bit weird here.

Reply:
I think it's reasonable. We need to explicit the ownership clearly here. (http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rr-unique)

Here scheduler is owned by gcs_server, unless it's a sharing semantic, which happens more naturally in sharing obj between threads, we'd like to keep it a unique ptr. cc @scv119

Reply:
Fixed.